src/sage/combinat/bijectionist.py

```diff
@@ -0,0 +1,3217 @@
+
+        """
+        Z = self._bijectionist._Z
+        Z_dict = {z: i for i, z in enumerate(Z)}
+        zero = self.milp.linear_functions_parent().zero()
+        for tA, tZ in self._bijectionist._elements_distributions:
+            tA_sum = [zero]*len(Z_dict)
+            tZ_sum = [zero]*len(Z_dict)
+            for a in tA:
+                p = self._bijectionist._P.find(a)
+                for z in self._bijectionist._possible_block_values[p]:
+                    tA_sum[Z_dict[z]] += self._x[p, z]
+            for z in tZ:
+                tZ_sum[Z_dict[z]] += 1
+
+            for a, z in zip(tA_sum, tZ_sum):
+                self.milp.add_constraint(a == z, name=f"d: {a} == {z}")
+
+    def add_intertwining_relation_constraints(self):
+        r"""
+        Add constraints corresponding to the given intertwining
+        relations.
+
+        INPUT:
+
+        - origins, a list of triples `((\pi/\rho, p,
+          (p_1,\dots,p_k))`, where `p` is the block of
+          `\rho(s(a_1),\dots, s(a_k))`, for any `a_i\in p_i`.
+
+        This adds the constraints imposed by
+        :meth:`set_intertwining_relations`,
+
+        .. MATH::
+
+            s(\pi(a_1,\dots, a_k)) = \rho(s(a_1),\dots, s(a_k))
+
+        for each pair `(\pi, \rho)`.  The relation implies
+        immediately that `s(\pi(a_1,\dots, a_k))` only depends on the
+        blocks of `a_1,\dots, a_k`.
+
+        The MILP formulation is as follows.  Let `a_1,\dots,a_k \in
+        A` and let `a = \pi(a_1,\dots,a_k)`.  Let `z_1,\dots,z_k \in
+        Z` and let `z = \rho(z_1,\dots,z_k)`.  Suppose that `a_i\in
+        p_i` for all `i` and that `a\in p`.
+
+        We then want to model the implication
+
+        .. MATH::
+
+           x_{p_1, z_1} = 1,\dots, x_{p_k, z_k} = 1 \Rightarrow x_{p, z} = 1.
+
+        We achieve this by requiring
+
+        .. MATH::
+
+            x_{p, z}\geq 1 - k + \sum_{i=1}^k x_{p_i, z_i}.
+
+        Note that `z` must be a possible value of `p` and each `z_i`
+        must be a possible value of `p_i`.
+
+        EXAMPLES::
+
+            sage: A = B = 'abcd'
+            sage: bij = Bijectionist(A, B, lambda x: B.index(x) % 2)
+            sage: def pi(p1, p2): return 'abcdefgh'[A.index(p1) + A.index(p2)]
+            sage: def rho(s1, s2): return (s1 + s2) % 2
+            sage: bij.set_intertwining_relations((2, pi, rho))
+            sage: from sage.combinat.bijectionist import _BijectionistMILP
+            sage: bmilp = _BijectionistMILP(bij)                                # indirect doctest
+            sage: next(bmilp.solutions_iterator(False, []))
+            {'a': 0, 'b': 1, 'c': 0, 'd': 1}
+        """
+        A = self._bijectionist._A
+        tZ = self._bijectionist._possible_block_values
+        P = self._bijectionist._P
+        for composition_index, pi_rho in enumerate(self._bijectionist._pi_rho):
+            pi_blocks = set()
+            for a_tuple in itertools.product(*([A]*pi_rho.numargs)):
```

Comment:
```suggestion
            for a_tuple in itertools.product(A, repeat=pi_rho.numargs):
```
I think the `repeat` argument is what you are after here.